File: leoz-node/src/main/java/org/deku/leoz/node/data/sync/v1/EntityStateMessage.java
package org.deku.leoz.node.data.sync.v1;

import java.io.Serializable;
import java.sql.Timestamp;

/**
 * Message sent by
 * - consumers to request entity updates
 * - publishers to indicate entity updates
 * Created by masc on 18.06.15.
 */
public class EntityStateMessage implements Serializable {
    private static final long serialVersionUID = -3506609875846947166L;
    private Class mEntityType;
    private Timestamp mTimestamp;

    /**
     * c'tor
     * @param entityType
     * @param timestamp
     */
    public EntityStateMessage(Class entityType, Timestamp timestamp) {
        mEntityType = entityType;
        mTimestamp = timestamp;
    }
    public EntityStateMessage() { }

    /**
     * Entity type this state message refers to
     * @return
     */
    public Class getEntityType() {
        return mEntityType;
    }

    /**
     * Latest entity timestamp
     * @return
     */
    public Timestamp getTimestamp() {
        return mTimestamp;
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/data/sync/v1/EntityStateMessage.kt
package org.deku.leoz.node.data.sync.v1

import java.io.Serializable
import java.sql.Timestamp

/**
 * Message sent by
 * - consumers to request entity updates
 * - publishers to indicate entity updates
 * Created by masc on 18.06.15.
 *
 * @param entityType Entity type this state message refers to
 * @param timestamp Latest entity timestamp
 */
class EntityStateMessage(
        val entityType: Class<*>? = null,
        val timestamp: Timestamp? = null
) : Serializable {

    companion object {
        private val serialVersionUID = -3506609875846947166L
    }
}
